fix #293894: crash on Space when editing custom fingering

MarcSabatella opened the pull request:
When editing fingering with a "User" text style applied and then hitting Space,
we try to create a new text element on the next note,
but the attempt fails because we cmdAddText() keys off the text style,
and there is no handling for user styles (not clear what element type should be created).
Unfortunately there are no further checks for this and eventually there is a crash
as we attempt to update the text toolbar on the non-existent text element.
I should note that Alt+Right does not crash, but the operation still fails for the same reason.

This commit addresses the underlying problem directly
by making sure we create the element correctly:
passing in the *default* Tid for the desired element type in cmdAddText(),
and then overriding the Tid afterwards if necessary.
Thus, we never fail to create the text and thereby avoid any ill effects.
As a bonus, we gain the ability to add texts with a style different from the default.

dmitrio95 commented on mscore/scoreview.cpp:
```diff
@@ -4129,6 +4130,8 @@ void ScoreView::cmdAddText(Tid tid)
             _score->select(s, SelectType::SINGLE, 0);
             _score->endCmd();
+            if (customTid != Tid::DEFAULT && customTid != tid)
+                  s->initTid(customTid);
```
Wouldn't it be better to move this code before the `endCmd()` call? Then `Score::update()` will be called with a correct `Tid` for the element already.

MarcSabatella replied:
I think you're right.  Will update.

MarcSabatella commented:
Updated as per discussion above.  Also added a call to updateInspector() in the path where we are navigating between existing texts rather than adding new ones, as this doesn't otherwise happen when typing "Space" in a fingering.

anatoly-os commented:
@MarcSabatella could you please rebase the PR?

MarcSabatella commented:
Rebased (and thanks for the merge that necessitated it!)
